### PR TITLE
Fix ZMove structure

### PR DESCRIPTION
**Pull request by VeteranPadgett**
Moved where a move becomes the zmove so that when the base move is something like UTurn or Feint the  zmove doesn't switch out/break protection/etc. Moved where the zmoves used is set so that if a Pokemon faints before attacking the player can still use a zmove later. Moved Water Shuriken because ocd.

### CODE REVIEW

**Comment by Fuzzysqurl** [comment]
why close? :(

**Comment by VeteranPadgett** [comment]
Thought of one more thing to test but needed to go get a haircut. Now there is merge conflict :(

**Comment by Fuzzysqurl** [comment]
I'll fix those

File: src/BattleServer/moves.cpp [hunk 1]
```diff
@@ -8163,14 +8176,15 @@ struct MMZSupernova : public MM
 
     static void zm(int, int, BS &b) {
         if(b.terrain != BS::PsychicTerrain) {
-            //No terrain extender cause you can only hold 1 item!
```
Comment (Fuzzysqurl) [inline src/BattleServer/moves.cpp]: Also Hackmons still only allows 1 item to be held... You can't remove a Z Crystal either

**Pull request merged**